### PR TITLE
Document new experimental HTTP transport based on Netty 4 and Project Reactor

### PR DESCRIPTION
**Pull request by reta**
### Description
Document new experimental HTTP transport based on Netty 4 and Project Reactor

### Issues Resolved
Closes https://github.com/opensearch-project/documentation-website/issues/5540

### Checklist
- [ ] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by hdhalter** [comment]
Thanks, @reta ! Is this ready for a doc review?

**Comment by reta** [comment]
> Thanks, @reta ! Is this ready for a doc review?

Yes @hdhalter , thank you!

File: _install-and-configure/configuring-opensearch/network-settings.md [hunk 1]
```diff
@@ -48,3 +48,12 @@ OpenSearch supports the following advanced network settings for transport commun
 - `transport.bind_host` (Static, list): Specifies an address or addresses to which an OpenSearch node binds to listen for incoming transport connections. 
 
 - `transport.publish_host` (Static, list): Specifies an address or addresses that an OpenSearch node publishes to other nodes for transport communication.
+
+## Selecting the transport
+
+The default OpenSearch transport is provided by `transport-netty4` module and internally uses Netty 4 engine for both internal TCP-based communication between nodes in the cluster as well as external HTTP-based communication with the clients. This communication is fully asynchronous and non-blocking. However, there are other transport plugins available that could be used interchangeably:
```
Comment (vagimeli) [inline _install-and-configure/configuring-opensearch/network-settings.md:54]:
```suggestion
The default OpenSearch transport is provided by the `transport-netty4` module and internally uses the [Netty 4](https://netty.io/) engine for both internal Transmission Control Protocol (TCP)-based communication between nodes in the cluster and external HTTP-based communication with the clients. This communication is fully asynchronous and nonblocking. However, there are other transport plugins available that could be used interchangeably:
```

Comment (vagimeli) [reply _install-and-configure/configuring-opensearch/network-settings.md:54]: @reta I made some copy edits and added a link to Netty. Please review.

File: _install-and-configure/configuring-opensearch/network-settings.md [hunk 2]
```diff
@@ -48,3 +48,12 @@ OpenSearch supports the following advanced network settings for transport commun
...
+
+Plugin | Description
+:---------- | :--------
+`transport-nio`    | The OpenSearch transport based on Java NIO. <br> Installation: `./bin/opensearch-plugin install transport-nio` <br> Configuration (using `opensearch.yml`): <br> `transport.type: nio-transport` <br> `http.type: nio-http-transport`
+`transport-reactor-netty4`    | The OpenSearch HTTP transport based on Project Reactor and Netty 4 (**experimental**) <br> Installation: `./bin/opensearch-plugin install transport-reactor-netty4` <br> Configuration (using `opensearch.yml`): <br> `http.type: reactor-netty4`
```
Comment (vagimeli) [inline _install-and-configure/configuring-opensearch/network-settings.md:59]: Should we include a link to the Project Reactor source?

**Review by vagimeli — APPROVED** [review]
LGTM, with minor copy edits

File: _install-and-configure/configuring-opensearch/network-settings.md [hunk 3]
```diff
@@ -48,3 +48,12 @@ OpenSearch supports the following advanced network settings for transport commun
 - `transport.bind_host` (Static, list): Specifies an address or addresses to which an OpenSearch node binds to listen for incoming transport connections. 
 
 - `transport.publish_host` (Static, list): Specifies an address or addresses that an OpenSearch node publishes to other nodes for transport communication.
+
+## Selecting the transport
+
+The default OpenSearch transport is provided by the `transport-netty4` module and internally uses the [Netty 4](https://netty.io/) engine for both internal TCP-based communication between nodes in the cluster and external HTTP-based communication with the clients. This communication is fully asynchronous and non-blocking. However, there are other transport plugins available that could be used interchangeably:
```
Comment (natebower) [inline _install-and-configure/configuring-opensearch/network-settings.md:54]:
```suggestion
The default OpenSearch transport is provided by the `transport-netty4` module and uses the [Netty 4](https://netty.io/) engine for both internal TCP-based communication between nodes in the cluster and external HTTP-based communication with clients. This communication is fully asynchronous and non-blocking. However, there are other transport plugins available that can be used interchangeably:
```

**Review by natebower — COMMENTED** [review]
@vagimeli Just a few minor changes. Thanks!

File: _install-and-configure/configuring-opensearch/network-settings.md [hunk 4]
```diff
@@ -48,3 +48,12 @@ OpenSearch supports the following advanced network settings for transport commun
 - `transport.bind_host` (Static, list): Specifies an address or addresses to which an OpenSearch node binds to listen for incoming transport connections. 
 
 - `transport.publish_host` (Static, list): Specifies an address or addresses that an OpenSearch node publishes to other nodes for transport communication.
+
+## Selecting the transport
+
+The default OpenSearch transport is provided by the `transport-netty4` module and uses the [Netty 4](https://netty.io/) engine for both internal TCP-based communication between nodes in the cluster and external HTTP-based communication with clients. This communication is fully asynchronous and non-blocking. However, there are other transport plugins available that сфт be used interchangeably:
```
Comment (natebower) [inline _install-and-configure/configuring-opensearch/network-settings.md:54]:
```suggestion
The default OpenSearch transport is provided by the `transport-netty4` module and uses the [Netty 4](https://netty.io/) engine for both internal TCP-based communication between nodes in the cluster and external HTTP-based communication with clients. This communication is fully asynchronous and non-blocking. However, there are other transport plugins available that can be used interchangeably:
```

Comment (reta) [reply _install-and-configure/configuring-opensearch/network-settings.md]: Sorry about that, slipped through :(

**Pull request merged**